File: Python/Car.py
# Import class Vehicle yang menjadi parent class
from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, plat_nomor, merk, tahun_produksi, warna, jumlah_kursi, jumlah_pintu):
        # Memanggil konstruktor dari parent class (Vehicle)
        super().__init__(plat_nomor, merk, tahun_produksi, warna)
        
        # Menambahkan atribut khusus untuk class Car
        self.__jumlah_kursi = jumlah_kursi
        self.__jumlah_pintu = jumlah_pintu

    # Getter methods untuk mendapatkan nilai atribut
    def get_jumlah_kursi(self):
        return self.__jumlah_kursi

    def get_jumlah_pintu(self):
        return self.__jumlah_pintu

    # Setter methods untuk mengubah nilai atribut
    def set_jumlah_kursi(self, jumlah_kursi):
        self.__jumlah_kursi = jumlah_kursi

    def set_jumlah_pintu(self, jumlah_pintu):
        self.__jumlah_pintu = jumlah_pintu

File: CPP/Car.cpp
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle
{
    private:
        int jumlah_pintu;
        int jumlah_kursi;
    public:
    Car(){
        
    }

    Car(int jumlah_pintu, string plat_nomor, string merk, string tahun_produksi, string warna) : Vehicle(string plat_nomor, string merk, string tahun_produksi, string warna){
        this->jumlah_pintu = jumlah_pintu;
        jumlah_kursi ="";
    }
    Car(int jumlah_pintu,int jumlah_kursi, string plat_nomor, string merk, string tahun_produksi, string warna) : Vehicle(string plat_nomor, string merk, string tahun_produksi, string warna){
        this->jumlah_pintu = jumlah_pintu;
        this->jumlah_kursi = jumlah_kursi;
    }

    void set_jumlah_pintu(int jumlah_pintu){
        this->jumlah_pintu = jumlah_pintu;
    }

    void set_jumlah_kursi(int jumlah_kursi){
        this->jumlah_kursi = jumlah_kursi;
    }

    string get_jumlah_pintu(){
        return jumlah_pintu;
    }
    int get_jumlah_kursi(){
        return jumlah_kursi;
    }
}